src/components/validation/resources/http.rs

```diff
@@ -225,26 +235,56 @@ fn spawn_output_http_server(
     codec: ResourceCodec,
     output_tx: mpsc::Sender<Vec<Event>>,
     task_coordinator: &TaskCoordinator<Configuring>,
+    runner_metrics: &Arc<Mutex<RunnerMetrics>>,
 ) {
     // This HTTP server will wait for events to be sent by a sink, and collect them and send them on
     // via an output sender. We accept/collect events until we're told to shutdown.
 
     // First, we'll build and spawn our HTTP server.
     let decoder = codec.into_decoder();
 
-    let (_, http_server_shutdown_tx) =
-        spawn_http_server(task_coordinator, &config, move |request| {
+    let (_, http_server_shutdown_tx) = spawn_http_server(
+        task_coordinator,
+        &config,
+        runner_metrics,
+        move |request, output_runner_metrics| {
+            if request.headers().contains_key("Content-Length") {
+                let bytes = request
+                    .headers()
+                    .get("Content-Length")
+                    .unwrap()
+                    .to_str()
+                    .unwrap();
+                debug!("HTTP server external output resource received {bytes} bytes.");
+            }
             let output_tx = output_tx.clone();
             let mut decoder = decoder.clone();
 
             async move {
+                let mut output_runner_metrics = output_runner_metrics.lock().await;
                 match hyper::body::to_bytes(request.into_body()).await {
                     Err(_) => StatusCode::INTERNAL_SERVER_ERROR.into_response(),
                     Ok(body) => {
                         let mut body = BytesMut::from(&body[..]);
                         loop {
                             match decoder.decode_eof(&mut body) {
-                                Ok(Some((events, _byte_size))) => {
+                                Ok(Some((events, byte_size))) => {
+                                    debug!("HTTP server external output resource decoded {byte_size} bytes.");
+
+                                    // Update the runner metrics for the received events. This will later
+                                    // be used in the Validators, as the "expected" case.
+                                    output_runner_metrics.received_bytes_total += byte_size as u64;
+                                    output_runner_metrics.received_events_total +=
+                                        events.len() as u64;
+
+                                    // TODO: I think we will need to encode the events received and get the value
+                                    //       of the buffer length, to get the expected value for `component_sent_event_bytes_total`
+                                    //       Unfortunately that means we'll have to pass the encoder down to here as well.
```

Comment:
Just flagging this for reviewers

Reply:
Why would we need to encode the events to get their estimated JSON size? That part doesn't make sense to me.

Reply:
yeah, you are right, this makes no sense.

I think this hair brained idea came about because, when I set the validators to look at the `runner_metrics` (expected values) that I _think_ they should be looking at for sinks (the reciprocals) , the numbers don't match up ... need to find the real reason why.

Reply:
got to the bottom of this- the expected values was wrong because when the input is a controlled edge, we need to add the log namespace metadata to the event.